drivers/lora/Kconfig.semtech

```diff
@@ -4,16 +4,29 @@
 # SPDX-License-Identifier: Apache-2.0
 #
 
-menuconfig LORA_SX1276
-	bool "Semtech SX1276 driver"
+menuconfig LORA_SX12XX
+	bool "Semtech SX-series driver"
 	select HAS_SEMTECH_RADIO_DRIVERS
 	select HAS_SEMTECH_LORAMAC
-	select HAS_SEMTECH_SX1276
 	depends on SPI
 	depends on COUNTER
+	help
+	  Enable LoRa driver for Semtech SX12xx.
+
+if LORA_SX12XX
+
+choice
+	prompt "LoRa Radio chipset"
+	default LORA_SX1276
```

Comment:
I think that we should have some automatic selection of the right driver based on dt compatible. With current implementation user will need to select LORA_SX126X explicitly, in addition to configured device-tree node.

Reply:
I'm not sure if I know how to implement that in a good way. We would need to build different source files based on DT properties, which I don't think is possible. We could of course use some weird include hack that pulls in a radio implementation depending on the DT, but that seems quite scary.

Reply:
> We would need to build different source files based on DT properties, which I don't think is possible.

It is. Please look in scripts/kconfig/kconfigfunctions.py. For example dt_compat_enabled() will let you detect if a devicetree node of a particular compatible is turned on. You can search for uses in the rest of the code for example uses.

Reply:
Cool. I'll make sure to use it in that case. Thanks for the pointer!

Reply:
NP. Note that since you can only access Kconfig values in CMake, and not dt properties, the workaround for now is to do something like

``
config MY_DRIVER_WHATEVER
    default y if $(dt_something_something,...)
```

then build your driver if CONFIG_MY_DRIVER_WHATEVER=y in CMake.

Reply:
I'm not super familiar with Kconfig, but I think I implemented this the "right" way. I ended up adding a choice of chip types and defaulting to the one in the DT.